### PR TITLE
fixes #4924 feat(nimbus): add UI for new risk mitigation fields

### LINKED ISSUE #4924: (FE) Add 3 Risk Mitigation questions to Overview UI, query + mutation
The risk questions can live in the FE. Add these questions and radio buttons to the Overview form per Figma.

Once EXP-1129 is complete the FE should be able to add the risk mitigation questions to the experiment query to receive the values in the FE saved by the user on that experiment, and display the radio button accordingly (yes/no or neither if null).

Add the fields to the FE experiment mutation (after EXP-1128 is complete) so they can be updated. Also add these fields to the FE `readyForReview` array so the user is required to answer before the review can be requested (this functionality may be changed later in the "Required for Launch Errors" epic).

"Learn more" links should be:

* Brand and Guidance: https://mana.mozilla.org/wiki/pages/viewpage.action?spaceKey=FIREFOX&title=Pref-Flip+and+Add-On+Experiments#PrefFlipandAddOnEx[…]eriments-VPSign-off
* Partner company https://mana.mozilla.org/wiki/display/FIREFOX/Pref-Flip+and+Add-On+Experiments#PrefFlipandAddOnExperiments-Isthisstudypartnerrelated?
* Revenue: https://mana.mozilla.org/wiki/display/FIREFOX/Pref-Flip+and+Add-On+Experiments#PrefFlipandAddOnExperiments-Doesthisstudyhavepossiblenegativeimpactonrevenue?

See Figma: https://www.figma.com/file/5SLa07XqQb0ttB1ay1gami/Nimbus-Console-UI?node-id=218778%3A23479

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-1131)
┆Issue Number: EXP-1131

### PR DESCRIPTION
**Pull request by jodyheavener**
Closes #4924

This PR:

- Adds the new UI for the 3 risk mitigation questions to the overview page
- Adds the 3 fields to the "missing fields" check (this won't do anything until #5169 is merged)

### CODE REVIEW

File: app/experimenter/nimbus-ui/src/components/PageEditOverview/index.test.tsx [hunk 1]
```diff
@@ -48,23 +49,34 @@ describe("PageEditOverview", () => {
...
+      riskRevenue: experiment.riskRevenue!,
+      riskPartnerRelated: experiment.riskPartnerRelated!,
+    };
+    mockSubmitData = {
+      ...mockMutationData,
+      ...{
+        riskBrand: String(experiment.riskBrand!),
+        riskRevenue: String(experiment.riskRevenue!),
+        riskPartnerRelated: String(experiment.riskPartnerRelated!),
+      },
```
Comment (jodyheavener) [inline app/experimenter/nimbus-ui/src/components/PageEditOverview/index.test.tsx:68]: This kinda sucks, but basically we submit a stringified boolean, and the GQL mutation needs to mock a real boolean.

File: app/experimenter/nimbus-ui/src/components/FormOverview/index.tsx [hunk 2]
```diff
@@ -49,10 +52,22 @@ export const overviewFieldNames = [
...
 type OverviewFieldName = typeof overviewFieldNames[number];
 
+const optionalBoolString = (
+  value: boolean | null | undefined,
+): string | null => {
+  if (value == null) {
+    return null;
+  }
+  return String(value);
+};
```
Comment (jodyheavener) [inline app/experimenter/nimbus-ui/src/components/FormOverview/index.tsx:69]: This and the below `optionalStringBool` are handling converting these fields' optional boolean values to and from strings. I tried seeing if react-hook-form had any magic that could do this for us (like `setValueAs`), but didn't see anything. Open to suggestions.

File: app/experimenter/nimbus-ui/src/components/PageEditOverview/index.tsx [hunk 3]
```diff
@@ -27,6 +27,10 @@ const PageEditOverview: React.FunctionComponent<PageEditOverviewProps> = () => {
   const refetchReview = useRef<() => void>();
   const [isServerValid, setIsServerValid] = useState(true);
 
+  const optionalStringBool = (value: string): boolean | null => {
+    return value.length ? value === "true" : null;
+  };
```
Comment (lmorchard) [inline app/experimenter/nimbus-ui/src/components/PageEditOverview/index.tsx:32]: Might be worth budding this off into a small library in lib/ along with `optionalBoolString` just for symmetry.

File: app/experimenter/nimbus-ui/src/components/FormOverview/index.tsx [hunk 4]
```diff
@@ -163,6 +181,52 @@ const FormOverview = ({
...
+              <Col sm={10} className="col-sm-10 pl-0">
+                <Form.Label className="mb-0">
+                  {isMissingField!("risk_brand") && (
+                    <span className="align-text-bottom mr-1">
+                      <InlineErrorIcon
+                        name="risk_brand"
+                        message="Please answer this question"
+                      />
+                    </span>
+                  )}
```
Comment (lmorchard) [inline app/experimenter/nimbus-ui/src/components/FormOverview/index.tsx:197]: Nit: In other fields, this icon shows up over to the right at the end of a label or input field. Should this follow that pattern?

File: app/experimenter/nimbus-ui/src/components/PageEditOverview/index.test.tsx (hunk 1 above)
Comment (lmorchard) [reply app/experimenter/nimbus-ui/src/components/PageEditOverview/index.test.tsx:68]: ISTR we also run into this in a few spots where we deal with a number as a string but then need to deal with it explicitly as a parsed number in GQL mocks. 🤔

**Review by lmorchard — APPROVED** [review]
LGTM with some scattered nits

File: app/experimenter/nimbus-ui/src/components/InputRadios/index.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    name="foo"
+    options={[
+      { label: "Yes", value: "true" },
+      { label: "No", value: "false" },
+    ]}
+    {...{ FormErrors, formControlAttrs }}
+  >
+    Howdy, neighbor
+  </InputRadios>
+);
```
Comment (jodyheavener) [inline app/experimenter/nimbus-ui/src/components/InputRadios/index.stories.tsx:34]: I think I'm missing something here as this story doesn't display, any ideas @lmorchard?

File: app/experimenter/nimbus-ui/src/components/FormOverview/index.tsx (hunk 2 above)
Comment (jodyheavener) [reply app/experimenter/nimbus-ui/src/components/FormOverview/index.tsx]: Converted to a library :)

> when is this value expected to be null vs undefined

Not sure, these are the types that were automatically generated from the GQL schema

> do we want the string "undefined" or to return literally undefined

I don't think we'd ever get there with this function as it performs a loose null check, so in this case an undefined value would return null

File: app/experimenter/nimbus-ui/src/components/FormOverview/index.tsx [hunk 6]
```diff
@@ -163,6 +181,52 @@ const FormOverview = ({
...
+                />
+                <Form.Check
+                  type="radio"
+                  value="false"
+                  label="No"
+                  className="ml-3"
+                  id="risk-brand-false"
+                  {...formControlAttrs("riskBrand", {}, false)}
+                />
+              </Col>
```
[the remark this answers is not in the document]
Comment (jodyheavener) [reply app/experimenter/nimbus-ui/src/components/FormOverview/index.tsx]: Moved it all to a component!

File: app/experimenter/nimbus-ui/src/components/PageEditOverview/index.test.tsx (hunk 1 above)
Comment (jaredlockhart) [reply app/experimenter/nimbus-ui/src/components/PageEditOverview/index.test.tsx:68]: @lmorchard I think I fixed all the string ints for ids already.  Why are we treating bools as strings?

Comment (jodyheavener) [reply app/experimenter/nimbus-ui/src/components/PageEditOverview/index.test.tsx:68]: The form radio buttons produce strings, but the server wants booleans. So in out tests we need to (well, we should to be accurate) mock submitting the form with strings and then convert them to bools once they hit the mutation

File: app/experimenter/nimbus-ui/src/components/InputRadios/index.stories.tsx (hunk 5 above)
Comment (jodyheavener) [reply app/experimenter/nimbus-ui/src/components/InputRadios/index.stories.tsx]: Nevermind, I was being silly :)

**Pull request merged**